pkg/api/http/handlers.go

```diff
@@ -33,47 +33,55 @@ type Handler = func(ctx context.Context, c *app.RequestContext)
 //	@Router			/v1/language/{router}/chat [POST]
 func LangChatHandler(routerManager *routers.RouterManager) Handler {
 	return func(ctx context.Context, c *app.RequestContext) {
-		var req *schemas.UnifiedChatRequest
+		// Unmarshal request body
+		var req []schemas.UnifiedChatRequest
```

Comment:
Essentially, the override would happen by passing a few `schemas.UnifiedChatRequest` in the payload. This means that the client has to duplicate the whole chat history N times to be able to do that which is a bit wasteful and blow off the memory blueprint as one request would require much more memory then it could have required to process the request.

Because of that I think it's better to have the payload look like this:

```json
{
  "message": {...},
  // default message for all non-overridden requests
  "messageHistory": [
    {...},
    {...}
  ],
  "override": {
    "modelId": {
      "message": "...",
      // possibly other params
    },
    "modelId": {
      "message": "...",
      // possibly other params
    }
  }
}
```

Reply:
The point of this is that the message and messageHistory are different across models. We might be able to get away with not duplicating messageHistory but we need to duplicate messages.

Reply:
>The point of this is that the message and messageHistory are different across models. We might be able to get away with not duplicating messageHistory but we need to duplicate messages.

@mkrueger12 I'm wondering how would that work on the application side. When I'm thinking about this, I imagine a situation where we have basically one conversation/chat like:
- Q: What is the capital of United States? [Glide routed that Q to OpenAI model] The capital is Washington DC.
- Q:  What's the population of Washington? [Glide routed that Q to Cohere model] The population is 700k people.
- Q: How many parks Washihgton has? [Glide routed that Q back to OpenAI model] More than 140 parks

Don't we have one chat history across all interactions that is not really connected to the specific model that gave these answers?

I want to get this better. I do understand the case where you may want to ask about parks, let's say, differently depending on the model type. But, how/why would an application assemble different chat histories depending on model? 🤔 

In this specific case, the OpenAI chat history would be:
- What is the capital of United States? The capital is Washington DC.
-  How many parks Washinton has? More than 140 parks

while Cohere History would be just one Q-A pair?
- What's the population of Washington? The population is 700k people.

Reply:
I think the chat history remains the same regardless of model. So we can have on chat history. I may have missed this earlier but your example above is the direction I think we go. 

- A single shared history with different messages per model.

@roma-glushko How does that sound?

Reply:
> A single shared history with different messages per model.

@mkrueger12 yes, that's perfect (at least as long I understand the needs 😃 )